sql/logic_test.go

```diff
@@ -201,28 +201,27 @@ func (t logicTest) run(path string) {
 			// blank line or a line of the form "xx values hashing to yyy". The
 			// latter format is used by sqllogictest when a large number of results
 			// match the query.
-			if !s.Scan() {
-				t.Fatalf("%d: expected query results line\n", s.line)
-			}
-			if m := resultsRE.FindStringSubmatch(s.Text()); len(m) == 3 {
-				var err error
-				query.expectedValues, err = strconv.Atoi(m[1])
-				if err != nil {
-					t.Fatal(err)
-				}
-				query.expectedHash = m[2]
-			} else {
-				for {
-					results := strings.Fields(s.Text())
-					if len(results) == 0 {
-						break
+			if s.Scan() {
+				if m := resultsRE.FindStringSubmatch(s.Text()); len(m) == 3 {
+					var err error
```

Comment:
Changed to `m != nil`. When using regexp matches you usually assume you know the indexes of the capturing groups.